components/SpotifyAuthorization/SpotifyAuthorization.c

```diff
@@ -0,0 +1,286 @@
+#include <stdio.h>
+#include "SpotifyAuthorization.h"
+#include "main.h"
+#include "JsonExtraction.h"
+
+bool FinishAthurisiation_FLG;
+
+SemaphoreHandle_t FindCodeSemaphore = NULL;
+SemaphoreHandle_t FailToFindCodeSemaphore = NULL;
+extern QueueHandle_t BufQueue1;
+extern SemaphoreHandle_t GetResponseSemaphore;
+
+static const char *TAG = "SpotifyTask";
+static const char *TAG_APP = "SPOTIFY";
+
+extern struct Token_ TokenParam;
+extern struct UserInfo_ UserInfo;
+
+void SpotifyAuth();
+static void SpotifyTask(void *pvparameters);
+
+/**
+ * @brief This function handles the first HTTPS request to Spotify and redirects the user to the authorization page.
+ *
+ * @param[in] req The HTTP request object.
+ * @return Returns ESP_OK if the request is processed successfully.
+ */
+static esp_err_t FirstRequest(httpd_req_t *req)
+{
+    char loc_url[SmallBuffer + 150];
+    printf("here send to client - Request_\n");
+    sprintf(loc_url, "http://accounts.spotify.com/authorize/?client_id=%s&response_type=code&redirect_uri=%s&scope=user-read-private%%20user-read-currently-playing%%20user-read-playback-state%%20user-modify-playback-state", ClientId, ReDirectUri);
+    httpd_resp_set_hdr(req, "Location", loc_url);
+    httpd_resp_set_type(req, "text/plain");
+    httpd_resp_set_status(req, "302");
+    httpd_resp_send(req, "", HTTPD_RESP_USE_STRLEN);
+    return ESP_OK;
+}
+/**
+ * @brief This function handles the callback HTTPS request from Spotify and processes the response data.
+ *
+ * @param[in] req The HTTP request object.
+ * @return Returns ESP_OK if the request is processed successfully.
+ */
+static esp_err_t HttpsUserCallBackFunc(httpd_req_t *req)
+{
+    char Buf[MediumBuffer];
+    if (httpd_req_get_url_query_str(req, Buf, sizeof(Buf)) == ESP_OK)
+    {
+        printf("\n\n\n%s\n\n", Buf);
+        httpd_resp_set_type(req, "text/plain");
+        httpd_resp_set_status(req, HTTPD_200);
+        httpd_resp_send(req, Buf, HTTPD_RESP_USE_STRLEN);
+        uint8_t a = FindCode(Buf, sizeof(Buf));
+        printf("\na=%d", a);
```

Comment:
is it a log? if yes replace with ESP_LOG and proper TAG

Reply:
I replaced with ESP_LOG